test/components/meta_test.dart

```diff
@@ -236,37 +262,91 @@ main() {
         ]
       }),
     );
+
+    int iterationCount = 0;
+
+    void runBreakingIterationTest(dynamic func, dynamic callback) {
+      iterationCount = 0;
+      func(lines, callback);
+      expect(iterationCount, 1, reason: func.toString());
+      iterationCount = 0;
+      func(multiLine, callback);
+      expect(iterationCount, 1, reason: func.toString());
+    }
+
     // Each Iterators
     // meta.segmentEach has been purposely excluded from this list
     // TODO fill out this list will all 'each' iterators
-    for (Function func in [geomEach]) {
-      // Meta Each function should only a value of 1 after returning `false`
-      // FeatureCollection
-      var count = 0;
-      func(lines, (
-        GeometryObject? currentGeometry,
-        int? featureIndex,
-        Map<String, dynamic>? featureProperties,
-        BBox? featureBBox,
-        dynamic featureId,
-      ) {
-        count += 1;
+    test('geomEach', () {
+      runBreakingIterationTest(geomEach, (geom, i, props, bbox, id) {
+        iterationCount += 1;
         return false;
       });
-      expect(count, 1, reason: func.toString());
-      // Multi Geometry
-      var multiCount = 0;
-      func(multiLine, (
-        GeometryObject? currentGeometry,
-        int? featureIndex,
-        Map<String, dynamic>? featureProperties,
-        BBox? featureBBox,
-        dynamic featureId,
-      ) {
-        multiCount += 1;
+    });
+
+    test('flattenEach', () {
+      runBreakingIterationTest(flattenEach, (feature, i, mI) {
+        iterationCount += 1;
         return false;
       });
-      expect(multiCount, 1, reason: func.toString());
-    }
+    });
+
+    test('propEach', () {
+      runBreakingIterationTest(propEach, (prop, i) {
+        iterationCount += 1;
+        return false;
+      });
+    });
+
+    test('featureEach', () {
+      runBreakingIterationTest(featureEach, (feature, i) {
+        iterationCount += 1;
+        return false;
+      });
+    });
+  });
+
+  test('flattenEach -- MultiPoint', () {
+    featureAndCollection(multiPoint.geometry!).forEach((input) {
+      List<GeometryObject?> output = [];
+      flattenEach(input, (currentFeature, index, multiIndex) {
+        output.add(currentFeature.geometry);
+      });
+      expect(output, [pt.geometry!, pt2.geometry!]);
+    });
+  });
+
+  test('flattenEach -- Mixed FeatureCollection', () {
+    List<Feature> features = [];
+    List<int> featureIndices = [];
+    List<int> multiFeatureIndicies = [];
+    flattenEach(fcMixed, (currentFeature, index, multiIndex) {
+      features.add(currentFeature);
+      featureIndices.add(index);
+      multiFeatureIndicies.add(multiIndex);
```

Comment:
we could also check for the type with `expect(foo, isA<Feature<Point>>());`

Reply:
did something close, since it wasn't type narrowing until I looked at the `.geometry` property